client/main.js

```diff
@@ -28,19 +28,32 @@ const ButtonUp = document.getElementById('ButtonUp');
 const ButtonDown = document.getElementById('ButtonDown');
 const ButtonLeft = document.getElementById('ButtonLeft');
 const ButtonRight = document.getElementById('ButtonRight');
+
 const Button = document.querySelectorAll(".Button");
 
 const ButtonPlayerSelector = document.getElementById('slider')
 
 // add a event listner to all the elements with the class button and on click on such element make it send data
+// for (let i = 0; i < Button.length; i++) {
+// 	Button[i].addEventListener("mousedown", (e) => {
+// 		console.log(parseInt(ButtonPlayerSelector.value) + 1)
+// 		const WsData = {
+// 			"player": parseInt(ButtonPlayerSelector.value) + 1,
+// 			"input": Button[i].id
+// 		}
+// 		// console.log(JSON.stringify(WsData))
+// 		ws.send(JSON.stringify(WsData))
+// 	});
+// };
+
 for (let i = 0; i < Button.length; i++) {
-    Button[i].addEventListener("mousedown", (e) => {
-        console.log(parseInt(ButtonPlayerSelector.value) + 1)
-        const WsData = {
-            "player": parseInt(ButtonPlayerSelector.value) + 1,
-            "input": Button[i].id
-        }
-        // console.log(JSON.stringify(WsData))
-        ws.send(JSON.stringify(WsData))
-    });
+	Button[i].addEventListener("click", (e) => {
```

Comment:
click only sends a signal when the mouse is released and not when pressed this could lead to a slight input delay